pkg/extensions/customresources.go

```diff
@@ -388,7 +394,10 @@ func WaitUntilExtensionObjectsMigrated(
 		return err
 	}
 
-	return flow.Parallel(fns...)(ctx)
+	if err := flow.Parallel(fns...)(ctx); err != nil {
```

Comment:
Nit: we can get rid of the if-err check,
```suggestion
	return gardencorev1beta1helper.DeprecatedDetermineError(flow.Parallel(fns...)(ctx))
```
Similarly for other occurences.